src/main/java/org/snomed/snowstorm/core/data/services/classification/RemoteClassificationServiceClient.java

```diff
@@ -76,11 +104,15 @@ String createClassification(String previousPackage, String dependencyPackage, Fi
 	}
 
 	ClassificationStatusResponse getStatus(String classificationId) {
-		TimerUtil getStatus = new TimerUtil("getStatus");
-		logger.debug("Fetching status for classification {}.", classificationId);
-		ClassificationStatusResponse classificationStatusResponse = restTemplate.getForObject("/classifications/{classificationId}", ClassificationStatusResponse.class, classificationId);
-		getStatus.finish();
-		return classificationStatusResponse;
+		ClassificationStatusResponse response = classificationStatuses.get(classificationId);
+		if (response == null) {
+			response = restTemplate.getForObject("/classifications/{classificationId}", ClassificationStatusResponse.class, classificationId);
+			if (response != null) {
+				classificationStatuses.put(response.getId(), response);
+			}
+		}
+
+		return response;
```

Comment:
When requesting the status of a classification, first check the internal map. If it is not there, then fall back to requesting directly from the classification service.

Reply:
I suggest we rename this method to `getStatusChange` and return null if there is no new status delivered via JMS. 

The `classificationStatuses` collection could be renamed to `classificationStatusChanges` and the status removed when `getStatusChange` is called. There would be no need to clear the collection on a scheduled basis. `ClassificationService` would need to be updated to deal with status changes and expect null values.